### ISSUE TITLE
Implement grafana / active monitoring of `atd-data03` and disk management

### ISSUE THREAD

**Issue by frankhereford** [cityofaustin/atd-data-tech#12804]
This is to track the initial discovery and testing of software packages to provide for some light active monitoring of atd-data03. Possible solutions include grafana and the goal is to get emails or SMS when something goes wrong, in an ideal world.

Also investigate solutions for pruning/trimming down docker images that accumulate.

**Comment by frankhereford** [comment]
# TLDR

* Get on the city network with the VPN or a 🚗 or however you do that
* `ssh -L 9876:localhost:9876 <yourhandle>@atd-data03.austintexas.gov`
* Browse to http://localhost:9876

# Details

This stack runs as a `docker compose` stack on `atd-data03` which is set to restart on boot. Grafana has a robust authentication mechanism, including support for OAuth and all that, but I've fully disabled it as we're going to use access to the machine itself via `ssh` as enough credential to view the data. This works because the machine is configured via `iptables` to drop all traffic that comes in except on specific ports, and 9876 isn't one of them.

The whole of the configuration is this:

```
version: '3'
services:
  node-exporter:
    image: prom/node-exporter:latest
    container_name: node-exporter
    network_mode: "host"
    pid: "host"
    volumes:
      - "/:/host:ro,rslave"
    command:
      - '--path.rootfs=/host'
      - '--web.listen-address=127.0.0.1:9100'

    restart: unless-stopped

  prometheus:
    image: ubuntu/prometheus:2.46.0-22.04_stable
    container_name: prometheus
    extra_hosts:
      - "host.docker.internal:host-gateway"
    environment:
      - TZ=America/Chicago
    network_mode: "host"
    volumes:
      - ./prometheus/prometheus.yaml:/etc/prometheus/prometheus.yml
    restart: unless-stopped

  grafana:
    image: grafana/grafana:latest
    container_name: grafana
    network_mode: "host"
    environment:
      - GF_AUTH_ANONYMOUS_ENABLED=true
      - GR_AUTH_ANONYMOUS_ORG_ROLE=Admin
      - GF_USERS_ALLOW_SIGN_UP=false
      - GF_SERVER_HTTP_PORT=9876
    volumes:
      - grafana_data:/var/lib/grafana
    restart: unless-stopped

volumes:
  grafana_data:
```

I'll ask y'all in our next sync if you want this in a github repo; I'm 50/50 on it.

**Comment by frankhereford** [comment]
I also added some documentation to gitbook with the above information, and I've requested a review to get it merged.

**Comment by mddilley** [comment]
@frankhereford I added a comment to my approval of your Gitbook doc edits, but I can't find where that went in Gitbook 🙃 so I'll add it here. Do we need a username/password to log in? I can navigate around the web app but I don't see the starred dashboard. Getting to the dashboard is a piece of 🍰 🙌 🚀
